code/datums/status_effects/debuffs.dm

```diff
@@ -944,3 +944,51 @@
 	name = "Electro-Magnetic Pulse"
 	desc = "You've been hit with an EMP! You're malfunctioning!"
 	icon_state = "hypnosis"
+
+/datum/status_effect/slimegrub
+	id = "grub_infection"
+	duration = 60 SECONDS //a redgrub infestation in a slime 
+	status_type = STATUS_EFFECT_UNIQUE
+	tick_interval = 1
+	alert_type = /atom/movable/screen/alert/status_effect/grub
+	var/adult = FALSE 
+	var/spawnbonus = 0
+	var/deathcounter = 300
+	var/list/diseases = list()
+
+/datum/status_effect/slimegrub/on_apply(mob/living/new_owner, ...)
+	. = ..()
+	if(isslime(new_owner))
+		var/mob/living/simple_animal/slime/S = new_owner
+		if(S.is_adult)
+			adult = TRUE
+			duration = world.time + 120 SECONDS
+			if(S.amount_grown >= 9)
+				S.amount_grown = 8 //can't split or evolve
+		deathcounter = (300 + (300 * adult))
+
+/datum/status_effect/slimegrub/tick()
+	if(isslime(owner))
+		var/mob/living/simple_animal/slime/S = owner
+		if(S.amount_grown >= 9)
+			S.amount_grown = 8
+		if((S.reagents.has_reagent(/datum/reagent/consumable/capsaicin) || S.bodytemperature > BODYTEMP_HEAT_DAMAGE_LIMIT) && prob(10)) //redgrubs don't like heat. heating them up for too long kills them
+			qdel(src)
+		else //don't tick while being cured
```

Comment:
Maybe make it so the counter doesn't increment and gibbing can't occur while sufficient "heat" is present?

Reply:
that's what this does

Reply:
From what I can tell, the else contains the gibbing part, and the else runs if the whole condition isn't met, and the whole condition includes a `prob(10)`. This would mean that 90% of the ticks where you have the cure, the deathcounter will still decrement, and if it hits 0, you will be gibbed. Am I missing something?

Reply:
ah, that's fair. lemme change this up

Reply:
fixed